src/pages/redirect-protection/index.js

```diff
@@ -0,0 +1,261 @@
+      // Return URL without scheme (e.g., "example.com/path" instead of "https://example.com/path")
+      return url.host + url.pathname + url.search + url.hash;
+    } catch {
+      return '';
+    }
+  },
+  render: ({ targetUrl, displayUrl }) =>
+    html`
+      <template layout="block overflow">
+        <style>
+          #bg {
+            position: absolute;
+            top: -250px;
+            left: 50%;
+            width: 1648px;
+            height: 1525px;
+            z-index: -1;
+            transform: translateX(-50%);
+          }
+
+          #c-1 {
+            position: absolute;
+            left: 300px;
+            top: 100px;
+            width: 800px;
+            height: 800px;
+            background: radial-gradient(
+              circle,
+              #a1e4ff 0%,
+              rgba(255, 255, 255, 0.1) 70%
+            );
+            opacity: 0.4;
+          }
+
+          #c-2 {
+            position: absolute;
+            left: 420px;
+            top: 320px;
+            width: 1200px;
+            height: 1200px;
+            background: radial-gradient(
+              circle,
+              #3751d5 0%,
+              rgba(255, 255, 255, 0.1) 65%
+            );
+            opacity: 0.3;
+          }
+
+          .container {
+            min-height: 100vh;
+            padding: 2rem 1.5rem 0;
+          }
+
+          .header {
+            color: var(--color-brand-secondary);
+          }
+
+          .modal {
+            background: var(--background-primary);
+            color: var(--color-primary);
+            border-radius: 16px;
+            padding: 32px 16px;
+            box-shadow: 0 4px 24px var(--shadow-panel);
+            max-width: 375px;
+            width: 100%;
+          }
+
+          .link {
+            color: var(--color-brand-secondary);
+            text-decoration: none;
+            word-break: break-all;
+          }
+
+          @media (min-width: 768px) {
+            .container {
+              padding: 2rem 2rem 0;
+            }
+
+            .modal {
+              padding: 40px 24px;
+              max-width: 600px;
+            }
+          }
+
+          @media (min-width: 1024px) {
+            .modal {
+              padding: 40px;
+              max-width: 720px;
+            }
+          }
+
+          @media (prefers-color-scheme: dark) {
+            #bg {
+              display: none;
+            }
+          }
+        </style>
+
+        <div id="bg">
+          <div id="c-1"></div>
+          <div id="c-2"></div>
+        </div>
+
+        <div class="container" layout="column items:center">
+          <div class="header" layout="margin:2:0">
+            <ui-icon name="logo-with-slogan" layout="height:3"></ui-icon>
+          </div>
+
+          <div class="modal" layout="column gap:2">
+            <ui-text type="headline-l" layout="block:center">
+              Tracking Redirect Alert
+            </ui-text>
+
+            ${targetUrl
```

Comment:
here use `html.resolve(target, ...)` if the `target` keeps result of the external function resolving your `targetUrl`, `displayUrl` and `hostname`